pkg/controller/helper/shared.go

```diff
@@ -52,18 +51,29 @@ func RetrieveTenantCR(providerAccount *ProviderAccount, client k8sclient.Client,
 }
 
 /*
-SetOwnersReference sets ownersReference in given object
+SetOwnersReference sets ownersReference in given object for Tenant
 - object
 - k8client
 - tenantCR
 */
 func SetOwnersReference(object controllerutil.Object, client k8sclient.Client, tenantCR *capabilitiesv1alpha1.Tenant) (bool, error) {
+	return SetOwnersReferenceByObjectAndMeta(object, client, tenantCR.ObjectMeta, tenantCR.TypeMeta)
+}
+
+/*
+SetOwnersReferenceByObjectAndMeta sets ownersReference in given object using ObjectMeta and TypeMeta
+- object
+- k8client
+- objectMeta
+- typeMeta
+*/
+func SetOwnersReferenceByObjectAndMeta(object controllerutil.Object, client k8sclient.Client, objectMeta metav1.ObjectMeta, typeMeta metav1.TypeMeta) (bool, error) {
```

Comment:
Not sure is this intended from the function but just want to note the current logic for these functions will overwrite any existing ownerReferences on the object.

It will also skip updating if the ownerReference lengths are the same. For example, if object had an ownerRef to `Tenant` but wanted to be updated to `DeveloperAccount` (don't believe there is this case yet), this will be skipped because the old and new ownerRef length of the same :thinking:

We might want to improve the latter at least :thinking: 

https://github.com/3scale/3scale-operator/blob/bd3283cffe3b04c0c57f7592a9d5dbb13c378cfb/pkg/controller/helper/shared.go#L70-L93

Reply:
Oh cool, didn't know that method existed :+1: 

Eh I think it's fine to continue using the current overwrite method for now as it does not cause any problems if the intention is to always just have the singular ownerReference :thinking: 

Yes, comparing lengths I think is valid if we're using the control runtime library method.

However I think a deeper comparison is needed for this one that is overwriting the ownerRefs, possibly by comparing the owner uid :thinking: